src/nameresolution/builtin.rs

```diff
@@ -103,6 +120,35 @@ fn define_string<'a>(cache: &mut ModuleCache<'a>) -> Type {
     Type::UserDefinedType(STRING_TYPE)
 }
 
+/// Defining the 'Ptr' built in type
+///
+/// type Ptr a = Ptr a
+fn define_ptr<'a>(cache: &mut ModuleCache<'a>) {
+    let location = Location::builtin();
+
+    let a = cache.next_type_variable_id(LetBindingLevel(0));
+    let ptr = Type::Primitive(PrimitiveType::Ptr);
+    
+    let name = "Ptr".to_owned();
+    let ptr_id = cache.push_type_info(name.clone(), vec![a], location);
+    assert_eq!(ptr_id, PTR_TYPE);
```

Comment:
Looks like this is the source of your type errors - you define a new `Ptr` type here which is distinct from the `PrimitiveType::Ptr` type. You should hopefully be able to remove these lines and the PTR_TYPE constant (or change `PTR_TYPE` to equal `Type::Primitive(PrimitiveType::Ptr)`)

Reply:
Thank you! It does make sense to just use the primitive type instead :)

Reply:
Since `Ptr` isn't defined as a keyword, how would I specify the type in scope? Can I get around supplying a `TypeInfoId` in 
```rust
    resolver.current_scope().types.insert("Ptr".to_string(), (something that is a TypeInfoId));
```

Reply:
Hmm, yeah I don't think we have a good way of importing that right now. Easiest way for now may be to make `Ptr` a keyword like `string` is